actionpack/lib/action_dispatch/middleware/debug_exceptions.rb

```diff
@@ -61,7 +61,7 @@ def render_exception(request, exception)
           begin
             content_type = request.formats.first
           rescue Mime::Type::InvalidMimeType
-            render_for_api_request(Mime[:text], wrapper)
+            return render_for_api_request(Mime[:text], wrapper)
```

Comment:
I have a slight preference for your other suggested fix in https://github.com/rails/rails/issues/38977:

```suggestion
            content_type = Mime[:text]
```

This will have the same result for API-only apps, but will still call `render_for_browser_request` for non-API apps.